dashboards-observability/public/components/event_analytics/explorer/log_patterns/patterns_tab.tsx

```diff
@@ -0,0 +1,333 @@
+    index: 'opensearch_dashboards_sample_data_logs',
+    ip: '218.148.135.12',
+    machine: '{"os":"win 8","ram":11811160064}',
+    memory: 'null',
+    message:
+      '218.148.135.12 - - [2018-07-22T04:18:12.345Z] "GET /beats/filebeat/filebeat-6.3.2-linux-x86_64.tar.gz_1 HTTP/1.1" 200 4531 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:6.0a1) Gecko/20110421 Firefox/6.0a1"',
+    phpmemory: 'null',
+    referer: 'http://www.opensearch-opensearch-opensearch.com/success/gemini-6a',
+    request: '/beats/filebeat/filebeat-6.3.2-linux-x86_64.tar.gz',
+    response: '200',
+    tags: 'warning',
+    timestamp: '2022-06-26 04:18:12.345',
+    url:
+      'https://artifacts.opensearch.org/downloads/beats/filebeat/filebeat-6.3.2-linux-x86_64.tar.gz_1',
+    utc_time: '2022-06-26 04:18:12.345',
+  };
+
+  const dummyTimeStampField = 'timestamp';
+
+  const dummyExplorerFields = {
+    availableFields: [
+      { name: 'agent', type: 'string' },
+      { name: 'bytes', type: 'long' },
+      { name: 'clientip', type: 'ip' },
+      { name: 'event', type: 'struct' },
+      { name: 'extension', type: 'string' },
+      { name: 'geo', type: 'struct' },
+      { name: 'host', type: 'string' },
+      { name: 'index', type: 'string' },
+      { name: 'ip', type: 'ip' },
+      { name: 'machine', type: 'struct' },
+      { name: 'memory', type: 'double' },
+      { name: 'phpmemory', type: 'long' },
+      { name: 'referer', type: 'string' },
+      { name: 'request', type: 'string' },
+      { name: 'response', type: 'string' },
+      { name: 'tags', type: 'string' },
+      { name: 'timestamp', type: 'timestamp' },
+      { name: 'url', type: 'string' },
+      { name: 'utc_time', type: 'timestamp' },
+    ],
+    queriedFields: [],
+    selectedFields: [{ name: 'message', type: 'string' }],
+    unselectedFields: [
+      { name: 'agent', type: 'string' },
+      { name: 'bytes', type: 'long' },
+      { name: 'clientip', type: 'ip' },
+      { name: 'event', type: 'struct' },
+      { name: 'extension', type: 'string' },
+      { name: 'geo', type: 'struct' },
+      { name: 'host', type: 'string' },
+      { name: 'index', type: 'string' },
+      { name: 'ip', type: 'ip' },
+      { name: 'machine', type: 'struct' },
+      { name: 'memory', type: 'double' },
+      { name: 'message', type: 'string' },
+      { name: 'phpmemory', type: 'long' },
+      { name: 'referer', type: 'string' },
+      { name: 'request', type: 'string' },
+      { name: 'response', type: 'string' },
+      { name: 'tags', type: 'string' },
+      { name: 'timestamp', type: 'timestamp' },
+      { name: 'url', type: 'string' },
+      { name: 'utc_time', type: 'timestamp' },
+    ],
+  };
+
+  const dummyRawQuery =
+    "source = opensearch_dashboards_sample_data_logs | where match(request,'filebeat')";
+
+  const emptyPattern = {
+    firstTimestamp: 0,
+    lastTimestamp: 0,
+    puncSignature: '',
+    patternName: '',
+    ratio: '',
+    count: '',
+  };
```

Comment:
Let's add a type to the pattern vairable. Type can be added in constants/explorer.ts